### DISCUSSION TITLE
Unable to disable file processor stats logging without also disabling metrics

### DISCUSSION THREAD

**Discussion by SamWheating** [apache/airflow#24271] [General] [answered]
### Apache Airflow version

2.3.2 (latest released)

### What happened

I recently disabled the printing of the file parsing stats by setting `scheduler.print_stats_interval` to `0` in the airflow config, because it was causing a lot of noise in the scheduler logs.  

However, this set of some monitors we have set up on the file processor, since we were no longer receiving `dag_processing.last_run.seconds_ago.<dagrun>` metrics.  

On closer inspection, it appears that these metrics are emitted at the same time that the file parsing stats are logged, so disabling the logs disabled the metrics:

https://github.com/apache/airflow/blob/1507ca48d7c211799129ce7956c11f4c45fee5bc/airflow/dag_processing/manager.py#L734-L739

https://github.com/apache/airflow/blob/1507ca48d7c211799129ce7956c11f4c45fee5bc/airflow/dag_processing/manager.py#L781-L787

### What you think should happen instead

I should be able to disable the logging of stats without also disabling the metrics.

### Operating System

Debian GNU/Linux 10 (buster)

### Deployment

Other 3rd-party Helm chart

### Anything else

Let me know if you have any preferences or suggestions for a possible fix.

I was thinking about adding an explicit `scheduler.print_file_processor_stats` boolean to decouple the printing of the stats from the computing of metrics. This way a user could enable or disable the printing of the metrics separately from each other (as the metrics can already be disabled via the `metrics.statsd_allow_list` configuration. 

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment, marked as answer]
We are about to vote (and possibly implement) https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-49+OpenTelemetry+Support+for+Apache+Airflow - I think this kind of feature belong there and it should be part of otel implementation. No matter what, such feature will not get out before 2.4, so even if it "seems" like simple flag, probably it is much better to implement it there, rather than invest in the old statsd interface that we are going to deprecate.

I will convert it to a discussion for now - but I would appreciate @SamWheating if you could see if the proposed OTEL support is something that woudl also address your needs (and will be far more configurable and versatile using existing/standard configuration/features of the OTEL components).

**Reply by SamWheating** [reply]
Hey Jarek - thanks for the quick response!

This sounds reasonable. I will follow along with the OpenTelemetry project and try to ensure that this issue gets addressed.
